fix and improve content on reference pages of table elements (#27958)
mdn/content: files/en-us/web/html/element/col/index.md

{{HTMLSidebar}}

> The **`<col>`** [HTML](/en-US/docs/Web/HTML) element defines one or more columns within a table and is used for defining common semantics among table columns. It may be used within a {{HTMLElement("colgroup")}} element that has no [`span`](/en-US/docs/Web/HTML/Element/colgroup#span) attribute defined.

Author: The first one reads clearer for me.

    >
    > If the table doesn't use a [`colspan`](/en-US/docs/Web/HTML/Element/td#colspan) attribute, use the `td:nth-of-type(an+b)` CSS selector. Set `a` to zero and `b` to the position of the column in the table, e.g. `td:nth-of-type(2) { vertical-align: middle; }` to center the second column cells vertically.
    >
    > If the table does use a [`colspan`](/en-US/docs/Web/HTML/Element/td#colspan) attribute, the effect can be achieved by combining adequate CSS attribute selectors like `[colspan=n]`, though this is not trivial.

- `width` {{deprecated_inline}}

  - : This attribute specifies a default width for each column. In addition to the standard pixel and percentage values, this attribute might take the special form `0*`, which means that the width of each column spanned should be the minimum width necessary to hold the column's contents. Relative widths such as `5*` can also be used. Note that this attribute overrides the specified `width` of its {{HTMLElement("colgroup")}} parent element.

    > **Note:** Do not use this attribute as it is deprecated (and not supported) in the latest standard: use the CSS {{cssxref("width")}} property instead.

## Usage notes

> - The `<col>` element may be used within a {{HTMLElement("colgroup")}} element that doesn't have a `span` attribute.
> - It's important to note that `<col>` elements do not structurally group columns together, as this is the role of the {{HTMLElement("colgroup")}} element.

Reviewer: ```suggestion
The `<col>` element is used within a {{HTMLElement("colgroup")}} element that doesn't have a `span` attribute. The `<col>` elements does not structurally group columns together. This is the role of the {{HTMLElement("colgroup")}} element.

Only a limited number of CSS properties affect columns:
-   {{cssxref("border")}}: The various `border` properties apply, but only if the `<table>` has {{cssxref("border-collapse", "border-collapse: collapse")}} set. 
- {{cssxref("background")}} : The various`background` properties will set the background for cells within the column, but will only be visible if the cell and row backgrounds are transparent. 
- {{cssxref("width")}}: The `width` property defines a minimum width for the column, as if {{cssxref("min-width")}} were set. 
- {{cssxref("visibility")}}: The {{cssxref("visibility")}} property is also supported.

```

In simpler terms, three "style columns" are defined, with the last two containing styling information across multiple columns of the table. To avoid starting with the first table column and thus applying styles to it, an empty `<col>` element is added, which by default spans one column, i.e. the first one.

```html
<table>
  <caption>
    Personal weekly activities
  </caption>
  <colgroup>
    <col />
    <col span="5" class="weekdays" />
    <col span="2" class="weekend" />
  </colgroup>
  <tr>
    <th>Period</th>
    <th>Mon</th>
    <th>Tue</th>
    <th>Wed</th>
    <th>Thu</th>
    <th>Fri</th>
    <th>Sat</th>
    <th>Sun</th>
  </tr>
  <tr>
    <th>a.m.</th>
    <td>Clean room</td>
    <td>Football training</td>
    <td>Dance Course</td>
    <td>History Class</td>
    <td>Buy drinks</td>
    <td>Study hour</td>
    <td>Free time</td>
  </tr>
  <tr>
    <th>p.m.</th>
    <td>Yoga</td>
    <td>Chess Club</td>
    <td>Meet friends</td>
    <td>Gymnastics</td>
    <td>Birthday party</td>
    <td>Fishing trip</td>
    <td>Free time</td>
  </tr>
</table>
```

> 
> ```css

Reviewer: ```suggestion

### CSS

We use CSS instead of deprecated HTML attributes to provide a background color to the columns and align the cell content:
```css
table {
  border-collapse: collapse;
}
caption {
  caption-side: bottom;
}
th, td {
  border: 1px solid rgb(140 140 140);
  text-align: center;
}
```

Reviewer: it's important to show the text-align being on the td and th.

.weekdays {
  background-color: #d7d9f2;
}

.weekend {
  background-color: #ffe8d4;
}
```

```css hidden

> table {
>   border-collapse: collapse;
>   border: 2px solid rgb(140, 140, 140);
>   font-family: sans-serif;
>   font-size: 0.8rem;
>   letter-spacing: 1px;
> }
> 
> caption {
>   caption-side: bottom;
>   padding: 10px;
> }
> 
> td,
> th {
>   border: 1px solid rgb(160, 160, 160);
>   padding: 8px 6px;
> }
> 
> td {
>   text-align: center;
> }

Reviewer: ```suggestion
table {
  font-family: sans-serif;
  font-size: 0.8rem;
  letter-spacing: 1px;
}
```

Reviewer: moved the "you should see this" stuff to be visible
